containers/src/Data/Graph.hs

```diff
@@ -610,19 +612,25 @@ preArr bnds      = tabulate bnds . preorderF
 postorder :: Tree a -> [a] -> [a]
 postorder (Node a ts) = postorderF ts . (a :)
 
-postorderF   :: Forest a -> [a] -> [a]
+postorderF   :: [Tree a] -> [a] -> [a]
 postorderF ts = foldr (.) id $ map postorder ts
 
 postOrd :: Graph -> [Vertex]
 postOrd g = postorderF (dff g) []
 
--- | A topological sort of the graph.
+-- | \(O(V+E)\). A topological sort of the graph.
 -- The order is partially specified by the condition that a vertex /i/
 -- precedes /j/ whenever /j/ is reachable from /i/ but not vice versa.
+--
+-- Note: A topological sort exists only when there are no cycles in the graph.
+-- If the graph has cycles, the output of this function will not be a
+-- topological sort. In such a case consider using 'scc'.
```

Comment:
It would be good to mention that `reverseTopSort` is (currently?) faster.

Reply:
Because of the `reverse`? That seems avoidable. We build the list from the tree so we should also be able to build the reverse of the list directly. I can make a PR later.